sqlite/src/func.c

```diff
@@ -796,11 +798,17 @@ static void comdb2CtxinfoFunc(
   if( sqlite3_value_type(argv[0])!=SQLITE_TEXT ){
     return;
   }
+
+  struct sql_thread *thd = pthread_getspecific(query_info_key);
+  struct sqlclntstate *clnt = thd!=NULL ? thd->clnt : NULL;
+
   zName = (const char *)sqlite3_value_text(argv[0]);
   if( sqlite3_stricmp(zName, "parallel")==0 ){
-    struct sql_thread *thd = pthread_getspecific(query_info_key);
-    struct sqlclntstate *clnt = thd!=NULL ? thd->clnt : NULL;
     sqlite3_result_int(context, clnt!=NULL && clnt->conns!=NULL);
```

Comment:
For parallel, I was thinking you could use the same "if (clnt) to wrap the sqlite3_result_int call...", e.g.:

```
if (clnt) {
  sqlite3_result_int(context, clnt->conns!=NULL);
}
```

Reply:
Skipping the call would result in NULL getting returned client, that, in addition to 0 and 1. Do you really want that? Also, it looks like clnt would never be null here.

Reply:
Yes, it would return a SQL NULL in the event that clnt is NULL.  Currently, as you point out, this should be "impossible"; however, that may not always be the case.